src/pages/redirects-interstitial.tsx

```diff
@@ -1,34 +1,41 @@
-import React from 'preact/compat'
+import React, { useEffect } from 'preact/compat'
+import { findOriginIsolationRedirect } from '../lib/path-or-subdomain.js'
+import { translateIpfsRedirectUrl } from '../lib/translate-ipfs-redirect-url.js'
+import RedirectPage from './redirect-page'
 
 /**
  * This page is only used to capture the ?helia-sw=/ip[fn]s/blah query parameter that
  * is used by IPFS hosted versions of the service-worker-gateway when non-existent paths are requested.
+ * This will only redirect if the URL is for a subdomain
  */
 export default function RedirectsInterstitial (): React.JSX.Element {
-  const windowLocation = translateIpfsRedirectUrl(window.location.href)
-  if (windowLocation.href !== window.location.href) {
-  /**
-   * We're at a domain with ?helia-sw=, we can reload the page so the service worker will
-   * capture the request
-   */
-    window.location.replace(windowLocation.href)
-  }
+  const [subdomainRedirectUrl, setSubdomainRedirectUrl] = React.useState<string | null>(null)
+  const [isSubdomainCheckDone, setIsSubdomainCheckDone] = React.useState<boolean>(false)
+  useEffect(() => {
+    async function doWork (): Promise<void> {
+      setSubdomainRedirectUrl(await findOriginIsolationRedirect(translateIpfsRedirectUrl(window.location.href)))
+      setIsSubdomainCheckDone(true)
+    }
+    void doWork()
+  })
 
-  return (<>First-hit on IPFS hosted service-worker-gateway. Reloading</>)
-}
+  useEffect(() => {
+    if (subdomainRedirectUrl != null && window.location.href !== subdomainRedirectUrl) {
+      /**
+       * We're at a domain with ?helia-sw=, we can reload the page so the service worker will
+       * capture the request
+       */
+      window.location.replace(subdomainRedirectUrl)
+    }
+  }, [subdomainRedirectUrl])
 
-/**
- * If you host helia-service-worker-gateway on an IPFS domain, the redirects file will route some requests from
- * `<domain>/<wildcard-splat>` to `https://<domain>/?helia-sw=<wildcard-splat>`.
- *
- * This function will check for "?helia-sw=" in the URL and modify the URL so that it works with the rest of our logic
- */
-function translateIpfsRedirectUrl (urlString: string): URL {
-  const url = new URL(urlString)
-  const heliaSw = url.searchParams.get('helia-sw')
-  if (heliaSw != null) {
-    url.searchParams.delete('helia-sw')
-    url.pathname = heliaSw
+  if (!isSubdomainCheckDone) {
+    return (<>First-hit on IPFS hosted service-worker-gateway. Determining state...</>)
   }
-  return url
+
+  if (subdomainRedirectUrl == null) {
+    return <RedirectPage />
+  }
+
+  return <>Waiting for redirect to subdomain...</>
```

Comment:
if we redirect to a subdomain, this page will not be rendered again for the requested URL... RedirectPage will render directly. This page should also *not render again* for any subsequent unique urls because the SW is registered and will trigger the redirect logic, which would then load RedirectPage if a "first-hit" for that subdomain.